.taskcluster.yml

```diff
@@ -1,215 +1,181 @@
-                  - {$eval: 'default_task_definition'}
-                  - {$eval: 'release_task_definition'}
-                  - $if: 'is_staging'
+        $if: 'tasks_for in ["github-pull-request", "github-release"] || tasks_for == "github-push" && head_branch == "refs/heads/master"'
+        then:
+          $let:
+            level:
+              $if: 'tasks_for in ["github-push", "github-release"] && repoUrl == "https://github.com/mozilla/application-services"'
+              then: '3'
+              else: '1'
+          in:
+            taskId: '${ownTaskId}'
+            taskGroupId: '${ownTaskId}'  # same as taskId; this is how automation identifies a decision task
+            schedulerId: '${trustDomain}-level-${level}'
+            created: {$fromNow: ''}
+            deadline: {$fromNow: '1 day'}
+            expires: {$fromNow: '1 year 1 second'}  # 1 second so artifacts expire first, despite rounding errors
+            metadata:
+              $merge:
+                - owner: "${ownerEmail}"
+                  source: '${repoUrl}/raw/${head_sha}/.taskcluster.yml'
+                - $if: 'tasks_for in ["github-push", "github-pull-request", "github-release"]'
+                  then:
+                    name: "Decision Task"
+                    description: 'The task that creates all of the other tasks in the task graph'
+            provisionerId: "aws-provisioner-v1"
+            workerType: "${trustDomain}-${level}-decision"
+            tags:
+              kind: decision-task
+            routes:
+              - checks
+            scopes:
+              # `https://` is 8 characters so, ${repoUrl[8:]} is the repository without the protocol.
+              $if: 'tasks_for == "github-push"'
+              then:
+                $let:
+                  short_head_branch:
+                    $if: 'head_branch[:11] == "refs/heads/"'
+                    then: {$eval: 'head_branch[11:]'}
+                    else: ${head_branch}
+                in:
+                  - 'assume:repo:${repoUrl[8:]}:branch:${short_head_branch}'
+              else:
+                $if: 'tasks_for == "github-pull-request"'
+                then:
+                  - 'assume:repo:github.com/${event.pull_request.base.repo.full_name}:pull-request'
+                else:
+                  $if: 'tasks_for == "github-release"'
+                  then:
+                    - 'assume:repo:${repoUrl[8:]}:release'
+
+            requires: all-completed
+            priority: lowest
+            retries: 5
+
+            payload:
+              env:
+                # run-task uses these to check out the source; the inputs
+                # to `mach taskgraph decision` are all on the command line.
+                $merge:
+                  - MOZILLA_BASE_REPOSITORY: '${baseRepoUrl}'
+                    MOZILLA_HEAD_REPOSITORY: '${repoUrl}'
+                    MOZILLA_HEAD_REF: '${head_branch}'
+                    MOZILLA_HEAD_REV: '${head_sha}'
```

Comment:
Looking at our other `taskgraph`'d projects, these environment variables are named according to the _group_ the project is in.
So, FFTV/fenix/AC named the variables `MOZILLA_`, rather than `FENIX_` or `ANDROID_COMPONENTS_`.
What group is "application services" in? According to github, the repo is a subset of the `mozilla` repo, so I assumed that that is the group? Or is "application services" in the "application services" group? Do you understand what I'm hung up on, and what do you think?

Reply:
> So, FFTV/fenix/AC named the variables `MOZILLA_`, rather than `FENIX_` or `ANDROID_COMPONENTS_`.

These projects are in the group `MOBILE_` https://github.com/mozilla-mobile/firefox-tv/blob/bcc1a9508e949c58f880976d94e6a11c2929dd59/.taskcluster.yml#L124 🙂

The group for application services is `APPSERVICES_` https://github.com/mozilla/application-services/pull/1945/files#diff-ac0229d1171c0983b27003af4c7441a5L128. I remember @MihaiTabara chose this one, but I don't remember the exact reason.